packages/state-transition/src/block/processWithdrawals.ts

```diff
@@ -0,0 +1,70 @@
+import {ssz, capella} from "@lodestar/types";
+import {MAX_EFFECTIVE_BALANCE, ETH1_ADDRESS_WITHDRAWAL_PREFIX, MAX_WITHDRAWALS_PER_PAYLOAD} from "@lodestar/params";
+
+import {CachedBeaconStateCapella} from "../types.js";
+import {decreaseBalance} from "../util/index.js";
+
+export function getExpectedWithdrawals(state: CachedBeaconStateCapella): capella.Withdrawal[] {
```

Comment:
It's imperative to add performance tests of this function. We must understand the cost for Lodestar given best case, average case and worst case. Please take as reference existing perf test cases or reach to me if doubts